src/Network/DNS.hs

```diff
@@ -161,7 +165,13 @@ queryRaw (Class cls) (Name name) qtype = withCResState $ \stptr -> do
                 unless (errno == eOK) $
                     throwErrno "res_query"
 
-                fail "res_query(3) failed"
+                h_errno <- c_get_h_errno
+                case h_errno of
+                  1 -> throw DnsHostNotFound
```

Comment:
Did you mean `throwIO` here and in other 3 cases below? `throwIO` is preferable in the IO context.